グループ作成・編集機能

Hiro3nori3 opened the pull request:
・Messageモデルを作った
・モデルのアソシエーションを記述した

Commit:
#WHAT Messageモデル作成とアソシエーション

・Messageモデルを作った
・モデルのアソシエーションを記述した

Commit:
#WHAT formタグをform_forに変更

・formタグをform_forに変更
・collection_check_boxの記述を追加
（この時点では、まだflash_message実装なし）

Commit:
#WHAT ルーティング設定

・ルーティングの記述をした（ネスト）

Commit:
#WHAT groups_controllerにグループ作成・編集機能を実装

・groups_controllerにグループの作成、編集機能を実装した
・この機会にedit.html.hamlファイル内のformタグもform_forに書き換えた

Commit:
#WHAT ルーティング修正

・rootのmessages#indexの部分をgroups#indexに変更した

Commit:
#WHAT サイドバーの作成

・サイドバーのビュー画面を作成した

Commit:
#WHAT　バリデーションとフラッシュメッセージ

・Groupモデルクラスにバリデーションの記述を加えた
・ビューにエラー時にメッセージがでるようにした

Commit:
#WHAT 間違えていじったファイル

・多分、無視して大丈夫だけど、コンフリクトがこわいのでコミット

ta910 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,16 +1,38 @@
 
-	def create
+	def edit
+		@group = Group.find(params[:id])
```
この記述、複数回出てるのでメソットにしてbefore_actionでよびましょう！

Hiro3nori3 replied:
おけ！

ta910 commented on app/views/groups/index.html.haml:
```diff
@@ -0,0 +1 @@
+=render partial: 'groups/sidebar', collection: @groups
```
render @groupsでうまくいきませんか？？

Hiro3nori3 replied:
いま部分テンプレートが_sidebar.html.hamlと_group.html.hamlの２つ存在するんだけど、これを_group.html.hamlだけでまとめるようにできないかってことかな？

ta910 commented on config/routes.rb:
```diff
@@ -1,7 +1,8 @@
 Rails.application.routes.draw do
-  devise_for :users
+	devise_for :users
```
インデント揃えましょう！

ta910 commented:
デフォルトのインデント、4になっていませんか？？2に直しましょう！

Hiro3nori3 commented:
ごめん、最近2にしたんだけど、前4だった名残でお見苦しいことになっています...

Commit:
#WHAT　修正 ver1

・routes.rbのインデント直し
・groups_controller内にset_groupを定義し、editアクションとupdateアクションの際に常に参照するようにした

ta910 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,16 +1,44 @@
 	end
 
 	def update
+		@group = Group.find(params[:id])
```
beforeで呼んでるので不要ですよ！

Hiro3nori3 replied:
おっと、うっかりミス

Commit:
#WHAT 修正 ver2

・index.html.hamlのcollectionの記述を消した
・groups_controllerのupdateアクション内の無駄な記述を消した
